stable/dask/templates/NOTES.txt

```diff
@@ -9,16 +9,60 @@ This release includes a Dask scheduler, {{ .Values.worker.replicas }} Dask worke
 
 The Jupyter notebook server and Dask scheduler expose external services to
 which you can connect to manage notebooks, or connect directly to the Dask
-cluster.   You can get these addresses by running the following:
+cluster. You can get these addresses by running the following:
+
+{{- if contains "NodePort" .Values.scheduler.serviceType }}
+
+  export DASK_SCHEDULER=$(kubectl get nodes --namespace {{ .Release.Namespace }} -o jsonpath='{.items[0].status.addresses[0].address}')
```

Comment:
If the goal is to be able to use this variable to auto-configure a Dask `Client`, the variable should be `DASK_SCHEDULER_ADDRESS`, as already set in:

https://github.com/helm/charts/blob/e3ba6f2b662e26c1fbcee189baffd61fb277de58/stable/dask/templates/dask-jupyter-deployment.yaml#L41-L42

Reply:
(Correct me if I'm wrong), but my understanding is that - when using NodePort or LoadBalancer for example, that service port 8786 in `values.yaml` is mapped to a port between 32000-32767 (default) by Kubernetes, which is the port value that the user would use in code or in a browser, outside of the cluster. [https://kubernetes.io/docs/concepts/services-networking/service/#nodeport](https://kubernetes.io/docs/concepts/services-networking/service/#nodeport)

If we were to just say to `echo tcp://${DASK_SCHEDULER_ADDRESS}` in `templates/NOTES.txt`, then the user would see something like `tcp://dask-scheduler:8786`

But with the changes in this PR to NOTES.txt, the user would then do
```
export DASK_SCHEDULER=$(kubectl get nodes --namespace default -o jsonpath='{.items[0].status.addresses[0].address}')

export DASK_SCHEDULER_PORT=$(kubectl get svc --namespace default test2-dask-scheduler -o jsonpath='{.spec.ports[?(@.name=="test2-dask-scheduler")].nodePort}')

echo tcp://$DASK_SCHEDULER:$DASK_SCHEDULER_PORT               -- Dask Client connection

>> tcp://xx.xx.x.xx:30401 -- Dask Client connection
```

And they would get the mapped/accessible port from kubernetes, something like `tcp://xx.xx.x.xx:30401    -- Dask Client connection` instead of `tcp://xx.xx.x.xx:8786` (DASK_SCHEDULER_ADDRESS) which wouldn't be accessible outside of the cluster.

Reply:
We can change what goes into the NOTES.txt file.  If it doesn't neatly fit then we can rework things.  

It looks like the `DASK_SCHEDULER` and `DASK_SCHEDULER_PORT` veriables used in that file are kind of doing their own thing separate from how Dask configuration environment variables are usually used.  Personally I don't think it matters much which variables we use there.

If we define a `DASK_SCHEDULER_ADDRESS` environment variable

Reply:
Sorry, misfire.

If we define a `DASK_SCHEDULER_ADDRESS` environment variable  ...

then it should include both the address and the port.  For other variables the helm chart can do what it likes, although note that these values will end up being inserted in to [Dask's config system](https://docs.dask.org/en/latest/configuration.html).

Reply:
Yes they are separate from the Dask configuration, these commands are just to figure out which public/external ports Kubernetes mapped the actual services to.

Reply:
Thanks for explaining the intent of these variables.  Because they start with `DASK_`, if they are set in the environment, the Dask config system will pick them up, but that won't matter as long as they don't overlap with incorrect values.  Name similarity is slightly confusing but so-be-it.

Reply:
I haven't tested this, but 1. since the scheduler/workers are launched as kubernetes pods, and 2. since these environment variables will be set by the user locally (outside of the pods) after installing the chart via copy/paste from the NOTES.txt output, I don't think these variables would be picked up by Dask.